src/Endatix.Api/Infrastructure/Cors/EndpointsCorsConfigurator.cs

```diff
@@ -0,0 +1,163 @@
+using Endatix.Framework.Hosting;
+using Microsoft.AspNetCore.Cors.Infrastructure;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Endatix.Api.Infrastructure.Cors;
+
+public class EndpointsCorsConfigurator(IOptions<CorsSettings> corsSettings, ILogger<EndpointsCorsConfigurator> logger, IAppEnvironment appEnvironment, IWildcardSearcher wildcardSearcher) : IConfigureOptions<CorsOptions>
+{
+    public const string ALLOW_ALL_POLICY_NAME = "AllowAll";
+
+    public const string DISALLOW_ALL_POLICY_NAME = "DisallowAll";
+
+    private readonly string[] _no_input_params = [];
+
+    public void Configure(CorsOptions options)
+    {
+        AddPredefinedPolicies(options);
+
+        var settingsValue = corsSettings?.Value;
+        var isDefaultPolicySet = false;
+
+        if (settingsValue?.CorsPolicies is { Count: > 0 } corsPolicies)
+        {
+            foreach (var policySetting in corsPolicies)
+            {
+                options.AddPolicy(
+                    policySetting.PolicyName,
+                    policy => BuildFrom(policy, policySetting)
+                );
+
+                if (settingsValue?.DefaultPolicyName?.Equals(policySetting.PolicyName) == true)
+                {
+                    options.DefaultPolicyName = policySetting.PolicyName;
+                    isDefaultPolicySet = true;
+                }
+            }
+
+            if (!isDefaultPolicySet)
+            {
+                options.DefaultPolicyName = corsPolicies[0].PolicyName;

```

Comment:
I don't understand this logic. It seems it none of the policies is the default, we put the first one as default.
If this is the desired logic, do we really need to set it? By default understand something set that is used when nothing is explicitly configured, not just picking the first.

Reply:
Correct, this was the behavior planned. Idea was that if someone has omitted the `DefaultPolicyName`, we will assign the first one as default. I agree, this can have side effects and can be tricky to understand leading to headbanging and frustration.

Considering, we always have default policies loaded for Development and Production environments, I will refactor to have the default fallback and the only one and simplify the logic instead of having 2 stage fallback. It will be simpler now: you either have set a correct default policy set or you use the built-in policy as fallback default poliy.